python/ray/data/context.py

```diff
@@ -227,6 +227,11 @@ def __init__(
         # The additonal ray remote args that should be added to
         # the task-pool-based data tasks.
         self._task_pool_data_task_remote_args: Dict[str, Any] = {}
+        # Max number of task failures that are allowed before aborting
```

Comment:
Can we call this max_allowed_block_failures or something similar? Since it is not actually tasks.

Also prefer the term "errors" over "failures" since failures is related to system-level exceptions. Actually, does this also apply to tasks that failed due to system-level exceptions? The comment should make it clear what counts.

Reply:
+1